code/game/objects/items/cosmetics.dm

```diff
@@ -131,60 +131,98 @@
 			to_chat(user, "<span class='warning'>[H] doesn't have a head!</span>")
 			return
 		if(location == BODY_ZONE_PRECISE_MOUTH)
-			if(!(FACEHAIR in H.dna.species.species_traits))
-				to_chat(user, "<span class='warning'>There is no facial hair to shave!</span>")
-				return
-			if(!get_location_accessible(H, location))
-				to_chat(user, "<span class='warning'>The mask is in the way!</span>")
-				return
-			if(H.facial_hair_style == "Shaved")
-				to_chat(user, "<span class='warning'>Already clean-shaven!</span>")
-				return
+			if(user.a_intent == INTENT_HELP)
+				if(H.gender == MALE)
+					if (H == user)
+						to_chat(user, "<span class='warning'>You need a mirror to properly style your own facial hair!</span>")
+						return
+					if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+						return
+					var/new_style = input(user, "Select a facial hair style", "Grooming")  as null|anything in GLOB.facial_hair_styles_list
+					if(!get_location_accessible(H, location))
+						to_chat(user, "<span class='warning'>The mask is in the way!</span>")
+						return
+					user.visible_message("<span class='notice'>[user] tries to change [H]'s facial hair style using [src].</span>", "<span class='notice'>You try to change [H]'s facial hair style using [src].</span>")
+					if(new_style && do_after(user, 60, target = H))
+						user.visible_message("<span class='notice'>[user] successfully changes [H]'s facial hair style using [src].</span>", "<span class='notice'>You successfully change [H]'s facial hair style using [src].</span>")
+						H.facial_hair_style = new_style
+						H.update_hair()
+
+					return

```

Comment:
what's this return for?

Reply:
Return after H.update_hair(), I guess I should add an extra tab here

Reply:
that makes more sense